packages/expo-cli/src/commands/upload/submission-service/android/ServiceAccountSource.ts

```diff
@@ -0,0 +1,73 @@
+import prompt from '../../../../prompt';
+import { existingFile } from '../../../../validators';
+import log from '../../../../log';
+
+enum ServiceAccountSourceType {
+  path,
+  prompt,
+  // credentialsService,
+  // ...
+}
+
+interface ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType;
+}
+
+interface ServiceAccountPathSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.path;
+  path: string;
+}
+
+interface ServiceAccountPromptSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.prompt;
+}
+
+export type ServiceAccountSource = ServiceAccountPathSource | ServiceAccountPromptSource;
+
+async function getServiceAccountAsync(source: ServiceAccountSource): Promise<string> {
+  switch (source.sourceType) {
+    case ServiceAccountSourceType.path:
+      return await handlePathSourceAsync(source);
+    case ServiceAccountSourceType.prompt:
+      return await handlePromptSourceAsync(source);
+  }
+}
+
+async function handlePathSourceAsync(source: ServiceAccountPathSource): Promise<string> {
+  if (!(await existingFile(source.path))) {
+    log.warn(`File ${source.path} doesn't exist.`);
+    return await getServiceAccountAsync({ sourceType: ServiceAccountSourceType.prompt });
+  }
+  return source.path;
+}
+
+async function handlePromptSourceAsync(_source: ServiceAccountPromptSource): Promise<string> {
+  const path = await askForServiceAccountPathAsync();
+  return await getServiceAccountAsync({
+    sourceType: ServiceAccountSourceType.path,
+    path,
+  });
+}
+
+async function askForServiceAccountPathAsync(): Promise<string> {
+  log(
+    'We need a Google Service Account JSON key to upload your app to Google Play Store.\n' +
+      "If you're not sure how to create one, learn more here: https://expo.fyi/creating-google-service-account"
+  );
+  const { filePath } = await prompt({
+    name: 'filePath',
+    message: 'Path to Google Service Account file:',
+    default: 'api-0000000000000000000-111111-aaaaaabbbbbb.json',
```

Comment:
after trying this out i think my mind has changed on this - the string looks a bit like an error. some options:

1) get rid of default value after all
2) look in cwd for something that matches this format, or is a json file and contains `"type": "service_account",` and suggest that one

Reply:
another option (and could be done in addition to it) would be that after the first submission of the app we save away the most recently used service account location to `.expo` and populate the default with that. but this may be overkill for a small thing, and it doesn't really help as much with the first submission

Reply:
I've been thinking about doing the same! It think it should be done in a follow-up PR though. I created an issue so we won't forget about it -  https://github.com/expo/expo-cli/issues/2166